### PR TITLE
Remove Repetitive Code 

### LINKED ISSUE #185: Ensure that the code samples are correct
A lot of stories feature repetitive code. To avoid duplication we generally use functions like
```js
export const heading = (level) => `<h${level}>Heading ${level}</h${level}>`
```
and then we write stories like 
```html
<Preview>
  <Story>{heading(1)}</Story>
</Preview>
```

This eliminates the code repetition but leads to incorrect code previews.
Screen Shot 2020-03-18 at 11 45 35

The correct code can be inserted into the preview using the `mdxSource` prop on the `Preview` component. So a better story would look like
```html
<Preview mdxSource={heading(1)}>
  <Story>{heading(1)}</Story>
</Preview>
```
An example can be found in PR #176.

The goal is to have this done across all stories in the storybook where such function-based code is used.

### PR DESCRIPTION
**Pull request by megha070**
## Fixes

Fixes #185  by @dhruvkb 

## Description
Removed repetitive code from cards.stories.mdx

## Checklist
 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the `master` branch of the repository.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by megha070** [comment]
image
@dhruvkb can you please help me rectify this error

**Comment by dhruvkb** [comment]
This error occurs if there is a blank line inside your story. MDX treats blank lines as paragraph breaks so it breaks your story into two parts.

**Comment by megha070** [comment]
@dhruvkb Can you please check my code, Maybe I am making some mistake in writing code? The error still persists.

File: src/stories/cards.stories.mdx [hunk 1]
```diff
@@ -3,26 +3,32 @@ import { figmaConfig } from './helpers'
...
+        <h4 class="card-title"><a href="#">This is a title</a></h4>
+        <span class="subtitle">12 Octobre 2019</span>
+        <div class="content">
+          Description metus ligula pharetra litora class massa viverra, facilisis cum aenean hendrerit eget magnis convallis.
+        </div>
+        <a href="#" class="button is-primary">Read more</a>
+    </div>
+  </article>
+</div>`
+
```
Comment (dhruvkb) [inline src/stories/cards.stories.mdx]: There are a lot of differences between PostHorizontal and PostVertical. This function is not serving both purposes correctly. Closely examine the differences in the HTML of both stories.

**Review by dhruvkb — CHANGES_REQUESTED** [review]
I know your work on the PR is not done, but I'll just point this bug now itself.

**Comment by dhruvkb** [comment]
@megha070 it's working on my computer now. Let me know if this fixes the issue on your machine as well.

**Comment by megha070** [comment]
Sure Thanks a ton.

**Comment by megha070** [comment]
Thank you so much @dhruvkb, It works on my system as well.

**Comment by megha070** [comment]
@dhruvkb Please let me know if there is scope of improvement and removing repetitive code further.

**Comment by megha070** [comment]
image
It was working fine a while ago, but now this error show up.

**Comment by megha070** [comment]
@dhruvkb rectified it, working fine on local.

**Review by dhruvkb — CHANGES_REQUESTED** [review]
Please resolve conflicts.

**Comment by megha070** [comment]
@dhruvkb Could you please tell me if I need to improvise this?

**Comment by zackkrida** [comment]
@dhruvkb I forgot to mention this since you missed the last sprint planning—could you revisit this PR this week? Whatever way we move forward with it, it would be nice to resolve. Thanks!

**Comment by dhruvkb** [comment]
Sure, this PR always falls out of my schedule. I'll resolve this today.

**Review by dhruvkb — APPROVED** [review]
Although code style could use some improvement, the changes in the PR are significant to overlook those for the time being.

**Pull request merged**